src/plugins/intel_cpu/src/nodes/col2im.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "col2im.h"
+#include "common/cpu_convert.h"
+#include "openvino/reference/col2im.hpp"
+
+#include <openvino/opsets/opset15.hpp>
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+Col2Im::Col2Im(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr context)
+    : Node(op, context, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)) {
+    const auto col2Im = ov::as_type_ptr<const ov::opset15::Col2Im>(op);
+    strides = col2Im->get_strides();
+    dilations = col2Im->get_dilations();
+    padsBegin = col2Im->get_pads_begin();
+    padsEnd = col2Im->get_pads_end();
+}
+
+void Col2Im::getSupportedDescriptors() {
+    // Validation is already done in the ov::opset15::Col2Im.
+}
+
+void Col2Im::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    ov::element::Type dataPrecision = getOriginalInputPrecisionAtPort(0);
+    ov::element::Type indexPrecision = getOriginalInputPrecisionAtPort(1);
+    ov::element::Type outputPrecision = getOriginalOutputPrecisionAtPort(0);
+
+    addSupportedPrimDesc(
+        {{LayoutType::ncsp, dataPrecision}, {LayoutType::ncsp, indexPrecision}, {LayoutType::ncsp, indexPrecision}},
+        {{LayoutType::ncsp, outputPrecision}},
+        impl_desc_type::ref);
+}
+
+bool Col2Im::created() const {
+    return getType() == Type::Col2Im;
+}
+
+bool Col2Im::needPrepareParams() const {
+    return false;
+}
+
+void Col2Im::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+template <class T, class T_idx>
+void Col2Im::executeImpl() {
+    const auto indexPrecision = getSrcMemoryAtPort(1)->getPrecision();
+    std::vector<T_idx> outputSizeVector(2);
+    cpu_convert(getSrcMemoryAtPort(1)->getData(),
```

Comment:
I attempted to pass `int32` data into the reference implementation, but the pointer then contains trash (high, seemingly random values). I'm still checking out why - it may be the reference works only with `int64`.

Reply:
Please verify the types mapping in the `OV_SWITCH` form `Col2Im::execute`, it can be related.

Reply:
It was, it's fixed now.